File: docs/guides/examples/cpp/src/data_ingestion_annotating.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client instance, then get or create a bucket
    auto client = IClient::Build("http://127.0.0.1:8383", {.api_token="my-token"});
    auto [bucket, create_err] = client->GetOrCreateBucket("my-bucket");
    assert(create_err == Error::kOk);

    // Send a record  with labels and content type
    IBucket::Time ts = IBucket::Time::clock::now();
    auto err = bucket->Write("entry-1", {
            .timestamp = ts,
            .labels = {{"name", "example"},
                       {"type", "simple"}},
            .content_type = "text/plain",
    }, [](auto rec) {
        rec->WriteAll("Some binary data");
    });
    assert(err == Error::kOk);

    return 0;
}

File: docs/guides/examples/py/src/data_ingestion_annotating.py
import time
import asyncio
from reduct import Client, Bucket


async def main():
    # Create a client instance, then get or create a bucket
    client = Client("http://127.0.0.1:8383", api_token="my-token")
    bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

    # Send a record  with labels and content type
    ts = time.time()
    await bucket.write(
        "entry-1",
        b"Some binary data",
        ts,
        labels={"name": "example", "type": "simple"},
        content_type="text/plain",
    )


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
